src/utils.ml

```diff
@@ -196,10 +196,7 @@ module Cached_digest = struct
               };
       digest
 
-  let remove fn =
-    match Hashtbl.find cache fn with
-    | None -> ()
-    | Some file -> file.timestamp_checked <- false
+  let remove fn = Hashtbl.remove cache fn
```

Comment:
Would you mind explaining this change? Why was setting it to false a problem before?

Reply:
- it was useless given that we were doing that for the targets of a rule that we always delete before running the rule anyway
- it didn't effectively removed entries from the on-disk cache